ably/types/connectiondetails.py

```diff
@@ -5,11 +5,17 @@
 class ConnectionDetails:
     connection_state_ttl: int
     max_idle_interval: int
+    connection_key: str
+    connection_id: str
 
-    def __init__(self, connection_state_ttl: int, max_idle_interval: int):
+    def __init__(self, connection_state_ttl: int, max_idle_interval: int,
+                 connection_key: str, connection_id: str):
         self.connection_state_ttl = connection_state_ttl
         self.max_idle_interval = max_idle_interval
+        self.connection_key = connection_key
+        self.connection_id = connection_id
```

Comment:
I just realised we made a mistake here - the `connectionId` is actually sent at the top level of the protocol message (ie not as part of `connectionDetails`). If you add an assertion to the test like `assert ably.connection.connection_details.connection_id` it will fail at the moment. I think the best solution is to add a new argument to `on_connected` and then store the `connection_id` as a new field on `ConnectionManager`.